src/BalconyBotanica/Hosts/Controllers/QuizController.cs

```diff
@@ -13,34 +16,71 @@ namespace BalconyBotanica.Hosts.Controllers
-                [FromQuery] int balcony_spaceSize,
-                [FromQuery] string plant_watering,
-                [FromQuery] string plant_function,
-                [FromQuery] string plant_toxicity)
+                [FromQuery, Required] string sunlight,
+                [FromQuery] int spaceSizeSquareMeters,
+                [FromQuery, Required] string wateringSchedule,
+                [FromQuery] string plantFunction,
+                [FromQuery] string toxicity)
         {
-            //TODO: with the questions filled in, 
-            // we generate a top 3, that fits the questions 
+            QuizAnswers quizAnswers = ParseQuizAnswers(sunlight, spaceSizeSquareMeters, wateringSchedule, plantFunction, toxicity);
             IEnumerable<PlantDataDbo> recommendedPlantsDbo = plantRepository.GetPlants();
             RecommendedPlants recommendedPlants = recommendedPlantsDbo.MapToRecommendedPlants();
 
-            return recommendedPlants;
+            return recommender.Recommend(recommendedPlants, quizAnswers);
         }
 
+
+
         [HttpGet("getPlantTopOne")]
-        public RecommendedPlants getPlantTopOne()
+        public RecommendedPlants GetPlantTopOne()
         {
-            //TODO: with the questions filled in, 
-            // we generate an plant
+            //TODO: for testing purpose. This one has no Path Query's yet.
 
             IEnumerable<PlantDataDbo> recommendedPlantDbo = plantRepository.GetPlantById(2);
             RecommendedPlants recommendedPlant = recommendedPlantDbo.MapToRecommendedPlants();
 
             return recommendedPlant;
         }
+
+        private static QuizAnswers ParseQuizAnswers(string sunlight, int spaceSizeSquareMeters, string wateringSchedule, string plantFunction, string toxicity)
+        {
+            if (!Enum.TryParse(sunlight, true, out Sunlight parsedSunlight))
+            {
+                // TODO: handle exeption if empty. Can't be empty
+                throw new Exception(nameof(sunlight) + " can't be empty.");
+            }
+
+            if (!Enum.TryParse(wateringSchedule, true, out WateringSchedule parsedWateringSchedule))
+            {
+                // TODO: handle exeption if empty. Can't be empty
+                throw new Exception(nameof(wateringSchedule) + " can't be empty.");
+
+            }
+
+            if (!Enum.TryParse(plantFunction, true, out PlantFunction parsedPlantFunction))
+            {
+
+                parsedPlantFunction = PlantFunction.NONE;
+            }
+
+            if (!Enum.TryParse(toxicity, true, out Toxicity parsedToxicity))
+            {
+                parsedToxicity = Toxicity.NONE;
+            }
+
+            QuizAnswers quizAnswers = new(
+                        sunlight: parsedSunlight,
+                        spaceSizeSquareMeters: spaceSizeSquareMeters,
+                        wateringSchedule: parsedWateringSchedule,
+                        plantFunction: parsedPlantFunction,
```

Comment:
Something to consider when testing, `plantFunction` can be null, does the above code in this file work if the input is the string "null"?

Reply:
I hope this catches that (with tests I'll know for sure :D)

```
 if (!Enum.TryParse(plantFunction, true, out PlantFunction parsedPlantFunction))
            {
                parsedPlantFunction = PlantFunction.NONE;
            }
```